packages/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts

```diff
@@ -313,7 +323,14 @@ export class AwsCustomResource extends cdk.Construct implements iam.IGrantable {
     } else {
       // Derive statements from AWS SDK calls
       for (const call of [props.onCreate, props.onUpdate, props.onDelete]) {
-        if (call) {
+        if (call?.assumedRole) {
```

Comment:
This would not work with `AwsCustomResourcePolicy.fromStatements()` for `policy` and `assumedRole` for a specific call?

Reply:
I am not entirely sure what you mean here. Are you talking about using both `policy` and `assumedRole` for the same call?

Reply:
No, not for the same call. It's just that this `if` is not reachable if `props.policy.statements.lentgh !== 0`. This means that if you want to assume a role only for a specific call and use statements for the others it won't work.

Reply:
Good point, @warrify?

Reply:
Hm ... I can't really see where this would be necessary. Given the use-case of cross-account calls there isn't really a way to do the other calls with only `policy`.

In the end the assumption was that this is only to be used if `policy` is not an option for one call, it isn't for any of the others either

Reply:
@jogold Does that ☝️  make sense, or should I handle the case where both are present?

Reply:
I think it's fine if we make clear in the docs that when using `fromStatements()` together with `assumedRole` the `sts:AssumeRole` actions should be included.

Reply:
> I think it's fine if we make clear in the docs that when using `fromStatements()` together with `assumedRole` the `sts:AssumeRole` actions should be manually included.

@eladb this has not been addressed (README update).

Reply:
@jogold Should be resolved now 👍